libmscore/timesig.cpp

```diff
@@ -215,6 +215,17 @@ void TimeSig::read(XmlReader& e)
             _sig.set(z1+z2+z3+z4, n);
             }
       _stretch.reduce();
+
+      // HACK: handle time signatures from scores before 3.5.0 differently on some special occasions.
+      // See https://musescore.org/node/308139.
+      if (score()->mscoreVersion() < QString("3.5.0")) {
```

Comment:
This works but I wonder if we have a more direct way of detecting this. Perhaps we need to consider adding a "Protocol version" (concept borrowed from Minecraft) which differs for every version and is an increasing integer to formulate the comparation between minor updates.